docs/do-not-merge.md

```diff
@@ -0,0 +1,809 @@
+
+- Type specializations including ARRAY typed columns
+- Better support for nested arrays of strings, longs, and doubles
+- Smarter index utilization
+
+`json` type columns created with Druid 28.0.0 are not backwards compatible with Druid versions older than 26.0.0.
+If you upgrade from one of these versions, you can continue to write nested columns in a backwards compatible format (version 4).
+
+For more information, see [Nested column format in the upgrade notes](#nested-column-format).
+
+[#14955](https://github.com/apache/druid/pull/14955)
+[#14456](https://github.com/apache/druid/pull/14456)
+
+# Additional features and improvements
+
+## SQL compatibility
+
+### Three-valued logic
+
+Druid native filters now correctly observe SQL [three-valued logic](https://en.wikipedia.org/wiki/Three-valued_logic#SQL), `true`, `false`, `unknown`, instead of Druid's classic two-state logic when you set the following configuration values:
+
+* `druid.generic.useThreeValueLogicForNativeFilters = true`
+* `druid.expressions.useStrictBooleans = true`
+* `druid.generic.useDefaultValueForNull = false`
+
+[#15058](https://github.com/apache/druid/pull/15058)
+
+### Strict booleans
+
+`druid.expressions.useStrictBooleans` is now enabled by default.
+Druid now handles booleans strictly using `1` (true) or `0` (false).
+Previously, true and false could be represented either as `true` and `false`, respectively, as well as `1` and `0`.
+
+If you don't explicitly configure this property in `runtime.properties`, clusters now use LONG types for any ingested boolean values, and in the output of boolean functions for transformations and query time operations.
+
+This change may impact your query results. For more information, see [SQL compatibility in the upgrade notes](#sql-compatibility-1).
+
+[#14734](https://github.com/apache/druid/pull/14734)
+
+### NULL handling
+
+`druid.generic.useDefaultValueForNull` is now disabled by default.
+Druid now differentiates between empty records, such as `' '`, and null records.
+Previously, Druid might treat empty records as empty or null.
+
+This change may impact your query results. For more information, see [SQL compatibility in the upgrade notes](#sql-compatibility-1).
+
+[#14792](https://github.com/apache/druid/pull/14792)
+
+## SQL-based ingestion
+
+### Azure connector
+
+Added support for Microsoft's Azure storage type.
+You can now use fault tolerance and durable storage with Microsoft Azure's blob storage.
+For more information, see [Durable storage](https://druid.apache.org/docs/latest/multi-stage-query/reference#durable-storage).
+
+[#14660](https://github.com/apache/druid/pull/14660)
+
+### New ingestion mode for arrays
+
+The MSQ task engine now includes an `arrayIngestMode` that determines how arrays are treated for ingestion. Previously, arrays were ingested as multi-value dimensions (MVDs). They can now be ingested as `ARRAY<STRING>` or numeric arrays instead if the ingest mode is set to `array`.
```

Comment:
```suggestion
The MSQ task engine now includes an `arrayIngestMode` that determines how string arrays are stored. MSQ till Druid 27 supported string arrays, and they were ingested as multi-value dimensions (MVDs). This is incorrect behavior,  and now they can now be ingested as string arrays as intended. This is incompatible with the existing behavior so please refer to the migration docs to learn more on how to migrate to the new behaviour.
```

Reply:
Ingestion mode primarily affects string arrays, and the release note should reflect that. Also, we should probably note that the default behavior is `mvd` instead of `array` (new behavior) and numeric arrays can only be ingested if `array` is set in the query context.

Reply:
@LakshSingla If MSQ supported string arrays until Druid 27, what happened in Druid 27? Did we stop supporting string arrays in Druid 27?

Reply:
Druid 27 and older incorrectly stored them as multi-value STRING columns instead of ARRAY<STRING>, without requiring wrapping the string array in `ARRAY_TO_MV` which is what should have been done. The docs I am working on should help clear this up

Reply:
Updated